### ISSUE TITLE
[A11Y] Triggering click on drawer button can cause layered backdrops

### ISSUE THREAD

**Issue by davwheat** [flarum/core#2663]
## Bug Report

**Current Behavior**
On mobile viewports, focusing the nav drawer toggle button and triggering a click on it multiple times (e.g. via space) causes multiple backdrops to be shown, but only one of these closes when closing the drawer.

![1l96mO](https://user-images.githubusercontent.com/7406822/110225227-44ac1d00-7edb-11eb-82b5-cf882fdb4304.gif)

**Steps to Reproduce**
1. Shrink viewport to mobile size
2. Use TAB to focus the drawer toggle button
3. Hit SPACE twice
4. Close the drawer
5. Backdrop remains

**Expected Behavior**
Only one backdrop should be created.

**Possible solution(s)**

I have two ideas:

1. We prevent this situation happening by checking if the drawer is currently open on button click. This fixes this situation.
2. We put a check in the Drawer class to prevent it being opened if App has `drawerOpen`. This fixes it in case any exts open the drawer or if something else can cause this. We may also wish to consider removing all `.drawer-backdrop` as a precautionary measure when closing, too.

**Comment by askvortsov1** [comment]
I'm assuming this is primarily problematic for accessibility devices? That's probably why we hadn't caught it yet. If so, this might be good to add to the mega-issue.

**Comment by davwheat** [comment]
Hehe I discovered it while working on a fix for #2565.

I'll add it now.
